lightning/restore/restore.go

```diff
@@ -333,15 +314,19 @@ func (rc *RestoreController) restoreSchema(ctx context.Context) error {
 			for _, tblMeta := range dbMeta.Tables {
 				tablesSchema[tblMeta.Name] = tblMeta.GetSchema(ctx, rc.store)
 			}
-			err = tidbMgr.InitSchema(ctx, dbMeta.Name, tablesSchema)
+			err = tidbMgr.InitSchema(ctx, rc.tidbGlue, dbMeta.Name, tablesSchema)
 
 			task.End(zap.ErrorLevel, err)
 			if err != nil {
 				return errors.Annotatef(err, "restore table schema %s failed", dbMeta.Name)
 			}
 		}
 	}
-	dbInfos, err := tidbMgr.LoadSchemaInfo(ctx, rc.dbMetas, rc.backend.FetchRemoteTableModels)
+	getTableFunc := rc.backend.FetchRemoteTableModels
+	if !rc.tidbGlue.OwnsSQLExecutor() {
```

Comment:
I think glue should provide `FetchRemoteTableModels` here

Reply:
at where lightning's glue is initiated, backend is not initiated. So I guess we could

- keep this implementation, because `OwnsSQLExecutor` has another usage that decide whether run `SET SQL_MODE =`
- decide backend then initiate glue
- initiate backend afterwards, and `glue.SetFetchRemoteTableModels`

what do you think